### PR TITLE
Improve outliers detection

### PR DESCRIPTION
**Pull request by langbart**
### Improving the identification of outliers in catches

- **Replaced mantle length (ML) and carapace length (CL) with total length (TL) in Octopus and Lobsters respectively. Now the weights of these groups seem much more reasonable.**
- **Median absolute deviation (MAD) for catch parameters is estimated depending on gear type.**

- **Added a new validation step consisting in identifying abnormal values in the weight-price relationship for each species** 
Species' weight-price relationship is mostly linear, the function `validate_price_weight` estimates Cook's distance for each weight-price linear model and adds an alert when  Cook's distance is higher than a certain threshold (see figure below). Although at first glance some values in red do not appear to be outliers, several observations potentially weigh a lot in the calculation of the final summaries and should be removed, like those highlighted in orange. With the current settings the method identified around 600 outliers.

plot_zoom_png

### CODE REVIEW

**Comment by efcaguab** [comment]
Just looking at the description of the changes this looks like a fantastic update!!!! I will finish checking tomorrow. 

Also thanks for attaching the plot. it's great to see the changes visually and makes it much easier to see the implications of the code

File: R/validation-functions.R [hunk 1]
```diff
@@ -280,13 +280,89 @@ validate_catch_params <- function(data,method=NULL, k_ind =NULL, k_length = NULL
...
+  alert_ids <-
+    dplyr::left_join(surveys_price_alerts, surveys_catch_alerts, by = "submission_id") %>%
+    dplyr::filter(.data$submission_id %in% single_catches) %>%
+    tidyr::unnest(.data$species_group) %>%
+    tidyr::unnest(.data$length_individuals) %>%
+    dplyr::select(.data$submission_id, .data$species, .data$total_catch_value, .data$weight) %>%
+    dplyr::filter(!is.na(.data$weight) & !is.na(.data$total_catch_value) & .data$weight != 0) %>%
+    dplyr::group_by(.data$species) %>%
+    dplyr::mutate(model = broom::augment(stats::lm(formula = .data$weight ~ .data$total_catch_value))) %>%
+    dplyr::mutate(cooksd = .data$model$`.cooksd`) %>%
```
Comment (efcaguab) [inline R/validation-functions.R:323]: Love this! Very elegant

**Review by efcaguab — CHANGES_REQUESTED** [review]
Great work Lorenzo. I'm really impressed. The idea of using the Cook's distance is brilliant and the code is great. 

I was only able to spot one potential problem with the code what should be relatively easy to fix:

- We need to update the code that handles and uploads the flags in Airtable so that the new one (17) is included. I think currently that's not done. Since the alert is recorded in both the price and the catch alert data frames, we need to make sure that the code still works OK (for example that flag is not updated twice or something like that). I'll leave it to you to decide what's the best solution, but an idea is that 1validate_price_weight` returns only one data frame with both the catch and price info. 

As a side question. How did you landed on 21 being the parameter for the threshold. Does this number appears to scale well for both "small" and "large" taxa?

**Comment by langbart** [comment]
Hey @efcaguab  happy that you like this pull. You are correct, the choice of 21 seemed to be a good compromise between the various taxa, even if I think the tolerance could be increased even more. I am not too happy with some observations that are reported as outliers and that do not appear to be. However, considering that they are relatively few on the total, together with the removal of several very "annoying" obs for the final estimates calculations, the net score seems positive to me.

I have resolved some conflicts and updated the missing part on Airtable. It seems to work, maybe give it a quick check if you can just to be safe.

**Comment by efcaguab** [comment]
Fantastic. I just put a minus sign you missed. I also noticed that there were duplicate flags in the Airtable but that might have been from previous runs because when I cleared and ran the code again everything looked peachy. 

Good stuff. Feel free to merge anytime.

**Review by efcaguab — APPROVED** [review]

**Pull request merged**